packages/ic-asset-certification/src/asset_router.rs

```diff
@@ -793,9 +841,10 @@ impl<'content> AssetRouter<'content> {
         content_type: Option<String>,
         encoding: Option<AssetEncoding>,
         range_begin: Option<usize>,
+        status_code: Option<HttpStatusCode>,
     ) -> AssetCertificationResult<(HttpResponse<'content>, HttpCertification)> {
         let mut content = asset.content;
-        let mut status_code = 200;
+        let mut status_code = status_code.unwrap_or(HttpStatusCode::Ok);
```

Comment:
This is the main change in this file. Everything else is just passing the status around.